analysis/tools/ccsh/src/main/kotlin/de/maibornwolff/codecharta/tools/ccsh/Ccsh.kt

```diff
@@ -75,15 +87,80 @@ class Ccsh : Callable<Void?> {
             commandLine.executionStrategy = CommandLine.RunAll()
-            if (args.isEmpty() || isParserUnknown(args, commandLine)) {
-                return executeInteractiveParser(commandLine)
+            return if (args.isEmpty()) {
+                val configuredParsers = offerInteractiveParserSuggestions(commandLine)
+                if (configuredParsers.isEmpty()) {
+                    return 0
+                }
+
+                return executeConfiguredParsers(commandLine, configuredParsers)
+            } else if (isParserUnknown(args, commandLine) || args.contains("--interactive") || args.contains("-i")) {
+                executeInteractiveParser(commandLine)
             } else {
-                return commandLine.execute(*sanitizeArgs(args))
+                commandLine.execute(*sanitizeArgs(args))
             }
         }
 
+        fun offerInteractiveParserSuggestions(commandLine: CommandLine): Map<String, List<String>> {
+            val inputFile: String = KInquirer.promptInput(
+                    message = "What is the name of the input file/folder/url?",
+                    hint = "path/to/input/")
+
+            val selectedParsers = ParserService.offerParserSuggestions(commandLine, PicocliParserRepository(), inputFile)
+
+            if (selectedParsers.isEmpty()) {
+                println(NO_USABLE_PARSER_FOUND_MESSAGE)
+                return emptyMap()
+            }
+
+            val shouldRunConfiguredParsers: Boolean =
+                    KInquirer.promptConfirm(
+                            message = "Do you want to run all configured parsers?",
+                            default = true)
+
+            if (!shouldRunConfiguredParsers) {
+                return emptyMap()
+            }
+
+            return (ParserService.configureParserSelection(commandLine, PicocliParserRepository(), selectedParsers))
+        }
+
+        fun executeConfiguredParsers(commandLine: CommandLine, configuredParsers: Map<String, List<String>>): Int {
+            val exitCode = AtomicInteger(0)
+            val threadPool = Executors.newFixedThreadPool(configuredParsers.size)
+            for (configuredParser in configuredParsers) {
+                threadPool.execute {
+                    val currentExitCode = executeConfiguredParser(commandLine, configuredParser)
+                    if (currentExitCode != 0) {
+                        println("Code: $currentExitCode")
+                        exitCode.set(currentExitCode)
+                    }
+                }
+            }
+            threadPool.shutdown()
+            threadPool.awaitTermination(1, TimeUnit.DAYS)
+
+            println("Code: ${exitCode.get()}")
+            if (exitCode.get() != 0) {
+                return exitCode.get()
+            }
+            // Improvement: Try to extract merge commands before so user does not have to configure merge args?
+            return ParserService.executeSelectedParser(commandLine, "merge")
```

Comment:
Is this call stable? Otherwise let's notify the users in a first step that certain .cc.json files have been created and that they can visualize or merge it by themselves.

Reply:
I am not sure if this is stable since I only did some basic testing with the merger, and in those tests this solution worked.

We could notify the user as you said and add that functionality with a bit more testing in a new issue/pr if that is okay with you?

Reply:
For now I will just print a message informing the user about the result and that he can manually use the merge-tool to combine all parser results